src/main/java/de/unijena/cheminf/mortar/model/fragmentation/algorithm/AlkylStructureFragmenter.java

```diff
@@ -0,0 +1,710 @@
+                    tmpSingleCarbonContainer.addAtom(tmpAtom);
+                    for (int i = 0; i < 3; i++) {
+                        PseudoAtom tmpPseudoAtom = new PseudoAtom();
+                        tmpSingleCarbonContainer.addAtom(tmpPseudoAtom);
+                        tmpSingleCarbonContainer.addBond(new Bond(tmpAtom, tmpPseudoAtom));
+                    }
+                } else if (tmpAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_QUATERNARY_CARBON_PROPERTY_KEY)) {
+                    tmpSingleCarbonContainer.addAtom(tmpAtom);
+                    for (int i = 0; i < 4; i++) {
+                        PseudoAtom tmpPseudoAtom = new PseudoAtom();
+                        tmpSingleCarbonContainer.addAtom(tmpPseudoAtom);
+                        tmpSingleCarbonContainer.addBond(new Bond(tmpAtom, tmpPseudoAtom));
+                    }
+                } else {
+                    tmpChainFragmentationContainer.addAtom(tmpAtom);
+                }
+            } else {
+                tmpRingFragmentationContainer.addAtom(tmpAtom);
+            }
+        }
+        //bond extraction
+        //superior performance compared to normal for iteration over Array length
+        for (IBond tmpBond : this.bondArray) {
+            if (tmpBond.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_FRAGMENTATION_PLACEMENT_KEY)) {
+                IAtom tmpBeginAtom = tmpBond.getBegin();
+                IAtom tmpEndAtom = tmpBond.getEnd();
+                boolean tmpIsBeginFragPlacement = tmpBeginAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_FRAGMENTATION_PLACEMENT_KEY);
+                boolean tmpIsEndFragPlacement = tmpEndAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_FRAGMENTATION_PLACEMENT_KEY);
+                boolean tmpIsBeginTertiary = tmpBeginAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_TERTIARY_CARBON_PROPERTY_KEY);
+                boolean tmpIsEndTertiary = tmpEndAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_TERTIARY_CARBON_PROPERTY_KEY);
+                boolean tmpIsBeginQuaternary = tmpBeginAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_QUATERNARY_CARBON_PROPERTY_KEY);
+                boolean tmpIsEndQuaternary = tmpEndAtom.getProperty(AlkylStructureFragmenter.INTERNAL_ASF_QUATERNARY_CARBON_PROPERTY_KEY);
+                if (tmpIsBeginFragPlacement && tmpIsEndFragPlacement) {
+                    if (!(tmpIsBeginTertiary || tmpIsEndTertiary || tmpIsBeginQuaternary || tmpIsEndQuaternary)) {
+                        tmpChainFragmentationContainer.addBond(tmpBond);
+                    }
+                }
+            }
+            else {
+                tmpRingFragmentationContainer.addBond(tmpBond);
+            }
+        }
+        //</editor-fold>
+        //
+        IAtomContainerSet tmpRingACSet = this.checkConnectivity(tmpRingFragmentationContainer);
+        IAtomContainerSet tmpSingleACSet = this.checkConnectivity(tmpSingleCarbonContainer);
```

Comment:
all fragments here are unconnected single carbons anyway, right? No need for a connectivity check, then.